tidy3d/components/medium.py

```diff
@@ -2927,6 +2927,61 @@ class CustomAnisotropicMediumInternal(CustomAnisotropicMedium):
     )
 
 
+class AbstractNonlinearMedium(AbstractMedium, ABC):
+    """Abstract nonlinear medium.
+
+    Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for large nonlinearities. Increasing `numiters` can help with convergence.
+
+    """
+
+    numiters: pd.PositiveInt = pd.Field(
+        1,
+        title="Number of iterations",
+        description="Number of iterations for solving nonlinear constitutive relation.",
+    )
+
+
+class KerrMedium(AbstractNonlinearMedium, Medium):
+    """Kerr medium described by an instantaneous chi3 nonlinear susceptibility, as in the
+    expression below for the nonlinear polarization.
+
+    Note
+    ----
+    .. math::
+
+        P_{NL} = \\epsilon_0 \\chi_3 |E|^2 E
+
+    Note
+    ----
+    The nonlinear constitutive relation is solved iteratively; it may not converge
+    for large nonlinearities. Increasing `numiters` can help with convergence.
+
+    Note
+    ----
+    For complex fields (e.g. when using Bloch boundary conditions), the nonlinearity
+    is applied separately to the real and imaginary parts, so that the above equation
+    holds when both E and :math:`P_{NL}` are replaced by their real or imaginary parts.
+    Since the physical fields are real, it wouldn't make sense to apply the instantaneous
```

Comment:
maybe instead of this last sentence with "it wouldn't make sense" we can soften it with something like "The nonlinearity is only applied to the real-valued fields since they are the physical fields."